Sources/Architecture/ARM64/ARM64BranchInfo.cpp

```diff
@@ -0,0 +1,130 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+#include "DebugServer2/Architecture/ARM64/Branching.h"
+
+#include <cstdlib>
+
+using namespace ds2::Architecture::ARM64;
+
+namespace {
+
+class ARM64Instruction {
+  uint32_t _insn;
+
+public:
+  ARM64Instruction(uint32_t insn = 0) : _insn(insn) {}
+
+  void reset(uint32_t insn = 0) { _insn = insn; }
+
+  static inline int64_t SignExt(uint64_t n, size_t size) {
+    size_t const sh = 64 - size;
+    return static_cast<int64_t>(n << sh) >> sh;
+  }
+
+  //
+  // B <imm>
+  // BL <imm>
+  // B.<cond> <imm>
+  //
+  inline bool getB(BranchInfo &info) const {
+    if ((_insn & 0x7c000000) == 0x14000000) {
```

Comment:
I don't understand how this catches the `b.<cond>` case:

- b <imm> [0b000101TTTTTTTTTTTTTTTTTTTTTTTTTT]
- bl <imm> [0b100101TTTTTTTTTTTTTTTTTTTTTTTTTT]
- b.<cond> <imm> [0b01010100TTTTTTTTTTTTTTTTTTT0CCCC]

Reply:
You're right, it doesn't. Oops.